IPython/core/page.py

```diff
@@ -100,7 +100,10 @@ def _detect_screen_size(screen_lines_def):
     # flags each time), we just save the initial terminal state and
     # unconditionally reset it every time.  It's cheaper than making
     # the checks.
-    term_flags = termios.tcgetattr(sys.stdout)
+    try:
+        term_flags = termios.tcgetattr(sys.stdout)
+    except termios.error as err:
+        raise TypeError
```

Comment:
Why turn this into a different exception that the caller will catch, instead of adding `termios.error` to what the caller catches?